contracts/IncentivePool.sol

```diff
@@ -152,14 +152,33 @@ contract IncentivePool is IIncentivePool {
     return (_incentiveIndex, _lastUpdateTimestamp);
   }
 
-  function withdrawResidue() external override {
-    require(msg.sender == _owner, 'onlyAdmin');
+  function withdrawResidue() external override onlyOwner {
     require(isClosed(), 'OnlyClosed');
     uint256 residue = IERC20(_incentiveAsset).balanceOf(address(this));
     IERC20(_incentiveAsset).safeTransfer(_owner, residue);
     emit IncentivePoolEnded();
   }
 
+  /**
+   * @notice Admin can update amount per second
+   */
+  function setAmountPerSecond(uint256 newAmountPerSecond) external override onlyOwner {
+    _incentiveIndex = getIncentiveIndex();
+
+    amountPerSecond = newAmountPerSecond;
+
+    emit RewardPerSecondUpdated(newAmountPerSecond);
+  }
+
+  /**
+   * @notice Admin can update incentive pool end timestamp
+   */
+  function setEndTimestamp(uint256 newEndTimestamp) external override onlyOwner {
```

Comment:
I think endTimestamp and AmountPerSecond are duplicated.
If admin set endTimeStamp to current time, user reward index could not be changed.
And If admin set AmountPerSecond to zero, user reward index also could not be changed.
So I wonder difference usages of these variables. How about using only endTimestamp?

Umm, Do you intend flexibility of endTimestamp and rewardPerSecond? So incentive length can be extended or incentive amount can be reduced. Right? Good approach!